daemon/containerd/image.go

```diff
@@ -103,9 +104,22 @@ func (i *ImageService) GetImage(ctx context.Context, refOrID string, options ima
 			if i.UpdatedAt.After(lastUpdated) {
 				lastUpdated = i.UpdatedAt
 			}
+			if isDanglingImage(i) {
+				if len(tagged) > 1 {
+					// This is unexpected - dangling image should be deleted
+					// as soon as another image with the same target is created.
+					// Log a warning, but don't error out the whole operation.
+					logrus.WithField("refs", tagged).Warn("multiple images have the same target, but one of them is still dangling")
+				}
+				continue
+			}
+
 			name, err := reference.ParseNamed(i.Name)
 			if err != nil {
-				return nil, err
+				// This is inconsistent with `docker image ls` which will
+				// still include the malformed name in RepoTags.
+				logrus.WithField("name", name).WithError(err).Error("failed to parse image name as reference")
+				continue
```

Comment:
https://github.com/moby/moby/blob/d0f6fa036b91a2e0776eae2bd2e2908488530d7d/daemon/containerd/image_list.go#L177
We can't pass the "bad" name here, because `References` is a `[]reference.Named` instead of `[]string` like in List...

Related discussion on `RepoTags` in list: https://github.com/moby/moby/pull/44840#discussion_r1084088210

IMO the inconsistency is worse than not showing the bad name in RepoTags, so maybe we should change the List behavior to match this?